forum/communications/views.py

```diff
@@ -1,3 +1,41 @@
 from django.shortcuts import render
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated
+from .models import Message
+from .serializers import ChatRoomSerializer, MessageSerializer
+from django.contrib.auth import get_user_model
 
-# Create your views here.
+User = get_user_model()
+
+
+def index(request):
+    return render(request, "communications/index.html")
+
+
+def room(request, room_name):
+    return render(request, "communications/room.html", {"room_name": room_name})
+
+
+class CreateConversationView(generics.CreateAPIView):
+    serializer_class = ChatRoomSerializer
+    permission_classes = [IsAuthenticated]
+
+    def perform_create(self, serializer):
+        serializer.save()
+
+
+class SendMessageView(generics.CreateAPIView):
+    serializer_class = MessageSerializer
+    permission_classes = [IsAuthenticated]
+
+    def perform_create(self, serializer):
+        serializer.save(sender=self.request.user)
```

Comment:
A user can potentially send a message to a room they are not part of.
Validate the room's participants before saving the message.

_Example:_

```
def perform_create(self, serializer):
    room = serializer.validated_data["room"]
    if self.request.user not in room.online.all():
        raise PermissionDenied("You are not a participant of this conversation.")
    serializer.save(user=self.request.user)
```